src/audio/module_adapter/module/generic.c

```diff
@@ -75,11 +75,12 @@ int module_load_config(struct comp_dev *dev, const void *cfg, size_t size)
 	return ret;
 }
 
-int module_init(struct processing_module *mod, const struct module_interface *interface)
+int module_init(struct processing_module *mod)
 {
 	int ret;
 	struct module_data *md = &mod->priv;
 	struct comp_dev *dev = mod->dev;
+	const struct module_interface *const interface = dev->drv->adapter_ops;
```

Comment:
this now seems to be the only location where you extract a pointer to the `interface` directly instead of going via the driver every time. I was proposing to make this change throughout the patch.